### PR TITLE
[Feature] Support more csv paramters for routine load

### PR DESCRIPTION
**Pull request by zaorangyang**
## What type of PR is this：
- [ ] BugFix
- [x] Feature
- [ ] Enhancement
- [ ] Refactor
- [ ] UT
- [ ] Doc
- [ ] Tool

## Which issues of this PR fixes ：

Fixes #

## Problem Summary(Required) ：

This PR supports more CSV attributes for routine load, just like stream load.Please reference to https://github.com/StarRocks/starrocks/pull/13582. One thing to note is that for routine load, we do not support  skipheader attribute.
For example, you can perform the CSV import with the following command for routine load.
```
CREATE ROUTINE LOAD test.example_tbl1_ordertest1 ON example_tbl1
COLUMNS TERMINATED BY ",",
COLUMNS (order_id, pay_dt, customer_name, nationality, temp_gender, price)
PROPERTIES
(
    "desired_concurrent_number" = "5",
    "trim_space"="true",
    "enclose"="\'",
    "escape"="\\",
)
FROM KAFKA
(
   "kafka_broker_list"="...",
   "kafka_topic"="test_csv",
   "kafka_partitions"="0,1,2,3,4,5",
   "property.kafka_default_offsets"="OFFSET_END"
   ...
);
```
## Checklist:

- [x] I have added test cases for my bug fix or my new feature
- [ ] This pr will affect users' behaviors
- [x] This pr needs user documentation (for new or modified features or behaviors)
  - [ ] I have added documentation for my new feature or new function

## Bugfix cherry-pick branch check:
- [ ] I have checked the version labels which the pr will be auto backported to target branch
  - [ ] 2.5
  - [ ] 2.4
  - [ ] 2.3
  - [ ] 2.2

### CODE REVIEW

**Review by rickif — DISMISSED** [review]

**Comment by zaorangyang** [comment]
run starrocks_fe_unittest

**Comment by wanpengfei-git** [comment]
#### [FE PR Coverage Check]

:disappointed: **fail** : 19 / 25 (76.00%)




#### file detail

|   |path|covered_line|new_line|coverage|not_covered_line_detail|
|----|----|----|----|----|----|
|:large_blue_circle:|com/starrocks/load/streamload/StreamLoadInfo.java|0|3|00.00%|[420, 421, 422]|
|:large_blue_circle:|com/starrocks/load/routineload/RoutineLoadJob.java|6|9|66.67%|[211, 217, 223]|
|:large_blue_circle:|com/starrocks/sql/ast/CreateRoutineLoadStmt.java|13|13|100.00%|[]|

File: fe/fe-core/src/main/java/com/starrocks/load/routineload/RoutineLoadJob.java [hunk 1]
```diff
@@ -206,6 +206,25 @@ public boolean isFinalState() {
     private static final String PROPS_JSONPATHS = "jsonpaths";
     private static final String PROPS_JSONROOT = "json_root";
 
+
+    public boolean isTrimspace() {
+        return trimspace;
+    }
+
+    protected boolean trimspace = false;
```
Comment (mofeiatwork) [inline fe/fe-core/src/main/java/com/starrocks/load/routineload/RoutineLoadJob.java:214]: how about reorder all fields before methods ?

File: fe/fe-core/src/main/java/com/starrocks/sql/ast/CreateRoutineLoadStmt.java [hunk 2]
```diff
@@ -485,6 +511,11 @@ public void checkJobProperties() throws UserException {
         } else {
```
Comment (chaoyli) [inline fe/fe-core/src/main/java/com/starrocks/sql/ast/CreateRoutineLoadStmt.java:511]: Are the format and parameter the same with the CSV stream load?

File: fe/fe-core/src/main/java/com/starrocks/load/routineload/RoutineLoadJob.java (hunk 1 above)
Comment (zaorangyang) [reply fe/fe-core/src/main/java/com/starrocks/load/routineload/RoutineLoadJob.java:214]: Updated

Comment (zaorangyang) [reply fe/fe-core/src/main/java/com/starrocks/load/routineload/RoutineLoadJob.java:214]: > how about reorder all fields before methods ?

Updated

File: fe/fe-core/src/main/java/com/starrocks/sql/ast/CreateRoutineLoadStmt.java (hunk 2 above)
Comment (zaorangyang) [reply fe/fe-core/src/main/java/com/starrocks/sql/ast/CreateRoutineLoadStmt.java:511]: > Are the format and parameter the same with the CSV stream load?

Good comment. I will change the name of trimspace to trim_space, just like stream load.

**Comment by wanpengfei-git** [comment]
#### [FE PR Coverage Check]

:disappointed: **fail** : 19 / 25 (76.00%)




#### file detail

|   |path|covered_line|new_line|coverage|not_covered_line_detail|
|----|----|----|----|----|----|
|:large_blue_circle:|com/starrocks/load/streamload/StreamLoadInfo.java|0|3|00.00%|[420, 421, 422]|
|:large_blue_circle:|com/starrocks/load/routineload/RoutineLoadJob.java|6|9|66.67%|[390, 394, 398]|
|:large_blue_circle:|com/starrocks/sql/ast/CreateRoutineLoadStmt.java|13|13|100.00%|[]|

**Review by chaoyli — DISMISSED** [review]

**Comment by wanpengfei-git** [comment]
#### [FE PR Coverage Check]

:disappointed: **fail** : 19 / 25 (76.00%)




#### file detail

|   |path|covered_line|new_line|coverage|not_covered_line_detail|
|----|----|----|----|----|----|
|:large_blue_circle:|com/starrocks/load/streamload/StreamLoadInfo.java|0|3|00.00%|[427, 428, 429]|
|:large_blue_circle:|com/starrocks/load/routineload/RoutineLoadJob.java|6|9|66.67%|[390, 394, 398]|
|:large_blue_circle:|com/starrocks/sql/ast/CreateRoutineLoadStmt.java|13|13|100.00%|[]|

**Review by meegoo — DISMISSED** [review]

**Review by chaoyli — DISMISSED** [review]

**Comment by wanpengfei-git** [comment]
run starrocks_admit_test

**Comment by zaorangyang** [comment]
run starrocks_admit_test

**Comment by zaorangyang** [comment]
run starrocks_admit_test

**Comment by wanpengfei-git** [comment]
#### [FE PR Coverage Check]

:disappointed: **fail** : 23 / 29 (79.31%)




#### file detail

|   |path|covered_line|new_line|coverage|not_covered_line_detail|
|----|----|----|----|----|----|
|:large_blue_circle:|com/starrocks/load/streamload/StreamLoadInfo.java|0|3|00.00%|[427, 428, 429]|
|:large_blue_circle:|com/starrocks/load/routineload/RoutineLoadJob.java|6|9|66.67%|[390, 394, 398]|
|:large_blue_circle:|com/starrocks/sql/ast/CreateRoutineLoadStmt.java|17|17|100.00%|[]|

**Review by chaoyli — APPROVED** [review]

**Review by meegoo — APPROVED** [review]

**Comment by wanpengfei-git** [comment]
run starrocks_admit_test

**Pull request merged**